examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts

```diff
@@ -2,32 +2,71 @@ import { Injectable } from '@nestjs/common';
 import { Witness } from '../../../models/witness';
 import { Proof } from '../../../models/proof';
 import { ICircuitService } from '../circuit.interface';
+import * as snarkjs from 'snarkjs';
+import * as verificationKey from '../../../../../../zeroKnowledgeKeys/circuit/circuit_verification_key.json';
+import 'dotenv/config';
 
 @Injectable()
 export class SnarkjsCircuitService implements ICircuitService {
-  public async createWitness(input: object): Promise<Witness> {
-    const witness: Witness = {
-      proof: { a: ['a'], b: [['b']], c: ['c'] },
-      publicInput: ['publicInput'],
-      verificationKey: 'verificationKey',
-    };
-
-    return witness;
+  public witness: Witness;
+
+  public async createWitness(inputs: object): Promise<Witness> {
+    this.witness = new Witness();
+
+    const proof = await this.createProof(inputs);
+    this.witness.proof = proof;
+
+    const publicInputs = await this.getPublicInputs(inputs);
+    this.witness.publicInputs = publicInputs;
+
+    this.witness.verificationKey = verificationKey;
+    return this.witness;
   }
 
-  public async createProof(witness: Witness): Promise<Proof> {
-    return witness.proof;
+  public async createProof(inputs: object): Promise<Proof> {
+    const { proof } = await this.executeCircuit(inputs);
+
+    const newProof = {
+      a: proof.pi_a,
+      b: proof.pi_b,
+      c: proof.pi_c,
+      protocol: proof.protocol,
+      curve: proof.curve,
+    } as Proof;
+
+    return newProof;
   }
 
-  public async verifyProof(proof: Proof, witness: Witness): Promise<boolean> {
-    return true;
+  public async verifyProofUsingWitness(witness: Witness): Promise<boolean> {
```

Comment:
@biscuitdey you need to specify the protocol and curve for proof verification ... rn it is only groth16 on the default bn254 but that might change.

Reply:
The protocol and curve have been added as params inside the proof model

```
export class Proof {
  a: string[];
  b: string[][];
  c: string[];
  protocol?: string;
  curve?: string;
}
```

Reply:
@biscuitdey ah yes, we need to then check for different. We still need to check what the protocol and curve is and then choose the correct prover system and curve. So that needs to be added.

Reply:
As we are using only groth16 for our simple use case, I think we can add the functionality for checking the different protocols and curves in a later PR.